### ISSUE TITLE
Don't allow facets with empty values

### ISSUE THREAD

**Issue by jvine** [sul-dlss/earthworks#644]
These are reported as violations of "Ensure link text is meaningful within context". In each case, it's an empty facet value that creates an empty but perceivable (and operable!) link for a screenreader, and a broken display otherwise. We should prevent empty facet values.

https://earthworks.stanford.edu/catalog/facet/dc_publisher_s?featured=census_data&q=
https://earthworks.stanford.edu/catalog/facet/dc_publisher_s?featured=geospatial_data&q=
https://earthworks.stanford.edu/catalog/facet/dc_format_s?featured=geospatial_data&q=
https://earthworks.stanford.edu/catalog/facet/dc_publisher_s?f[dct_provenance_s][]=Harvard&featured=california_data&q=

**Comment by jkeck** [comment]
The main complication for this is that we don't (currently) "own" all of the indexing code for getting content into EarthWorks (in particular the content that we index from OpenGeoMetadata via [GeoCombine rake tasks](https://github.com/sul-dlss/earthworks/blob/39879e6be6d6fa8abc5b99468479ad50bb620931/lib/tasks/earthworks.rake#L143-L147)).

The solr configs are now live, and we'll need to go through the process of re-indexing content to get rid of all the blank values.

* Our Stanford content should be fairly easy to remediate (and the move from XSLT -> Traject should hopefully address any blank values that were coming through there, but we'll need to validate that).
* The OGM content is relatively straight forward enough to re-index, and I'll go through that process now that the configs are live.
* The content that was indexed via OGP will be a bit harder to re-index as the indexer is not idempotent (it mints new "IDs" every indexing run) so that will require a more complex strategy for re-indexing (which there is a path forward on).

Once we get through the three re-indexes/validation steps above we should hopefully be rid of any lingering blank facets.
